core/error.rs

```diff
@@ -692,17 +692,24 @@ pub(crate) fn has_call_site(
 const DATA_URL_ABBREV_THRESHOLD: usize = 150;
 
 pub fn format_file_name(file_name: &str) -> String {
-  abbrev_file_name(file_name).unwrap_or_else(|| file_name.to_string())
+  abbrev_file_name(file_name).unwrap_or_else(|| {
+    // same as to_percent_decoded_str() in cli/util/path.rs
+    match percent_encoding::percent_decode_str(file_name).decode_utf8() {
+      Ok(s) => s.to_string(),
+      // when failing as utf-8, just return the original string
+      Err(_) => file_name.to_string(),
+    }
+  })
 }
 
 fn abbrev_file_name(file_name: &str) -> Option<String> {
-  if file_name.len() <= DATA_URL_ABBREV_THRESHOLD {
-    return None;
-  }
   let url = Url::parse(file_name).ok()?;
   if url.scheme() != "data" {
     return None;
   }
+  if file_name.len() <= DATA_URL_ABBREV_THRESHOLD {
+    return Some(file_name.to_string());
+  }
```

Comment:
This may slightly degrade performance with checking url scheme at first, but I think it's generally fine and improves readability.

Reply:
Is this necessary? If not I would suggest to revert this, parsing URL for every stack frame will have measureable impact

Reply:
My understanding of the role of the `abbrev_file_name` function is returning `Some` when the argument of the `file_name` string  starts with `data:...` due to avoid decoding Data URLs in the `format_file_name` function.
So, I think that it can rewrite as follows.

```rust
fn abbrev_file_name(file_name: &str) -> Option<String> {
  if !file_name.starts_with("data:") {
    return None;
  }
  if file_name.len() <= DATA_URL_ABBREV_THRESHOLD {
    return Some(file_name.to_string());
  }
  let url = Url::parse(file_name).ok()?;
 ...
```

Reply:
Yeah, that sounds good!

Reply:
I've fix it as above.
Thank you for point out the performance!